tests/protocol/mocks.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+import contextlib
+import re
+from azurelinuxagent.common.protocol.wire import WireProtocol
+from azurelinuxagent.common.utils import restutil
+from tests.tools import patch
+from tests.protocol.mockwiredata import WireProtocolData
+
+
+@contextlib.contextmanager
+def mock_wire_protocol(mock_wire_data_file):
```

Comment:
moved here from the previous mock_wire_protocol.py, with a few changes I am highlighting below